src/main/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigClientImpl.java

```diff
@@ -75,49 +75,34 @@ private FirebaseRemoteConfigClientImpl(Builder builder) {
   }
 
   @VisibleForTesting
-  String getRcSendUrl() {
-    return rcSendUrl;
+  String getRemoteConfigUrl() {
+    return remoteConfigUrl;
   }
 
   @VisibleForTesting
   HttpRequestFactory getRequestFactory() {
     return requestFactory;
   }
 
-  @VisibleForTesting
-  HttpRequestFactory getChildRequestFactory() {
-    return childRequestFactory;
-  }
-
   @VisibleForTesting
   JsonFactory getJsonFactory() {
     return jsonFactory;
   }
 
   @Override
   public RemoteConfigTemplate getTemplate() throws FirebaseRemoteConfigException {
-    HttpRequestInfo request = HttpRequestInfo.buildGetRequest(rcSendUrl)
+    HttpRequestInfo request = HttpRequestInfo.buildGetRequest(remoteConfigUrl)
             .addAllHeaders(COMMON_HEADERS);
     IncomingHttpResponse response = httpClient.send(request);
     RemoteConfigTemplate parsed = httpClient.parse(response, RemoteConfigTemplate.class);
 
     List<String> etagList = (List<String>) response.getHeaders().get("etag");
-
-    if (etagList == null || etagList.isEmpty()) {
-      throw new FirebaseRemoteConfigException(
-              ErrorCode.INTERNAL,
-              "ETag header is not available in the server response.", null, null,
-              RemoteConfigErrorCode.INTERNAL);
-    }
+    checkArgument(!(etagList == null || etagList.isEmpty()),
+            "ETag header is not available in the server response.");
 
     String etag = etagList.get(0);
-
-    if (Strings.isNullOrEmpty(etag)) {
-      throw new FirebaseRemoteConfigException(
-              ErrorCode.INTERNAL,
-              "ETag header is not available in the server response.", null, null,
-              RemoteConfigErrorCode.INTERNAL);
-    }
+    checkArgument(!Strings.isNullOrEmpty(etag),
```

Comment:
Here too. I'd even recommend adding a `getETag(IncomingResponse resp)` helper method, and doing all the validation there.